sdk/storage/src/blob/blob/mod.rs

```diff
@@ -66,6 +66,52 @@ create_enum!(RehydratePriority, (High, "High"), (Standard, "Standard"));
 
 create_enum!(PageWriteType, (Update, "update"), (Clear, "clear"));
 
+pub mod cr64_optional {
+    use serde::{self, Deserialize, Deserializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<crate::ConsistencyCRC64>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: Option<String> = Option::deserialize(deserializer)?;
+        trace!("s == {:?}", s);
+
+        Ok(match s {
+            None => None,
+            Some(s) => match s.len() {
+                0 => None,
+                _ => Some(
+                    crate::ConsistencyCRC64::decode(s)
+                        .map_err(|e| serde::de::Error::custom(e.to_string()))?,
+                ),
+            },
+        })
+    }
+}
+
+pub mod md5_optional {
+    use serde::{self, Deserialize, Deserializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<crate::ConsistencyMD5>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: Option<String> = Option::deserialize(deserializer)?;
+        trace!("s == {:?}", s);
+
+        Ok(match s {
+            None => None,
+            Some(s) => match s.len() {
+                0 => None,
+                _ => Some(
+                    crate::ConsistencyMD5::decode(s)
```

Comment:
This line is the only thing different between the two implementations. I think it's fine to keep the separate but we could consider combining them.

Reply:
Yes, I even wrote a singe function initially. 
But I didn't want to bang my head looking for the right serde incantation so I cheated 😃 ...